sdk/python/lib/pulumi/resource.py

```diff
@@ -160,13 +160,28 @@ def __init__(self,
                         self._providers = {**self._providers, pkg: provider}
 
         if not custom:
-            providers = opts.providers
-            if providers is not None:
-                self._providers = {**self._providers, **providers}
+            providers = self.__convert_providers(opt.provider, opts.providers)
```

Comment:
I don't think this is going to work, since the function is named `__convert_providers__`

Reply:
fixed.